L/LAMMPS/build_tarballs.jl

```diff
@@ -75,37 +93,69 @@ cmake -C ../cmake/presets/most.cmake -C ../cmake/presets/nolib.cmake ../cmake -D
     -DPKG_MOLECULE=ON \
     -DPKG_REPLICA=ON \
     -DPKG_SHOCK=ON \
-    -DLEPTON_ENABLE_JIT=no
+    -DLEPTON_ENABLE_JIT=no \
+    -DPKG_GPU=${GPU_OPTION} \
+    -DGPU_API=cuda \
+    ${cmake_extra_args} \
 
 make -j${nproc}
 make install
 
-if [[ "${target}" == *mingw* ]]; then
+if [[ "${bb_full_target}" == *mingw* ]]; then
     cp *.dll ${prefix}/bin/
 fi
+
+if [[ "${bb_full_target}" != *cuda\+none* ]]; then
+    unlink $prefix/cuda/lib/libcuda.so
+fi
 """
 
 augment_platform_block = """
     using Base.BinaryPlatforms
+
+    module __CUDA
+        $(CUDA.augment)
+    end
+
     $(MPI.augment)
-    augment_platform!(platform::Platform) = augment_mpi!(platform)
+
+    function augment_platform!(platform::Platform)
+        augment_mpi!(platform)
+        __CUDA.augment_platform!(platform)
+    end
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 # platforms = supported_platforms(; experimental=true)
 platforms = supported_platforms()
 platforms = expand_cxxstring_abis(platforms)
+cuda_platforms = expand_cxxstring_abis(CUDA.supported_platforms(min_version=v"11.0"))
+# Cmake toolchain breaks on aarch64, so only x86_64 for now
+filter!(p -> arch(p)=="x86_64", cuda_platforms)
+mpi_platforms, mpi_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.3.1", OpenMPI_compat="4.1.6, 5")
+cudampi_platforms, cudampi_dependencies = MPI.augment_platforms(cuda_platforms; MPItrampoline_compat="5.3.1", OpenMPI_compat="4.1.6, 5")
+
+all_platforms = [platforms; cuda_platforms; mpi_platforms; cudampi_platforms]
+for platform in all_platforms
+    # Only for the non-CUDA platforms, add the cuda=none tag, if necessary.
+    if !haskey(platform, "cuda")
```

Comment:
```suggestion
    if CUDA.is_supported(platform) && !haskey(platform, "cuda")
```

`cuda=none` can only be used when there's also a CUDA-enabled build, which isn't the case on Windows.

Reply:
See https://github.com/JuliaPackaging/Yggdrasil/issues/6911. The problem is that we can't generate binaries for Windows, because `nvcc` isn't a cross compiler. If you only provide a `cuda=none` artifact, that'll prevent use of your artifact when the user has CUDA installed, which is probably not what you want. Instead, we should only use the `cuda=none` platform tag when there's _also_ a non-`none` version, so that users always get an artifact.